### PR TITLE
Backport support for Alpine musl

### LINKED ISSUE #5: Backport Portola to jdk8u (for Alpine musl build support)
We are taking the best of the Bellsoft and Azul patches to their jdk8u distributions (a big thank you to them!) and applying that here for testing with the Adoptium community before Bellsoft submits the patch to OpenJDK proper.

### PR DESCRIPTION
**Pull request by calderast**
Apply [Alpine's patches](https://git.alpinelinux.org/aports/tree/community/openjdk8) with the addition of some small fixes for Alpine musl build support on jdk8u

Test build here: https://ci.adoptopenjdk.net/job/jdk8-alpine-linux-x64-hotspot/32/

### CODE REVIEW

File: hotspot/src/os/linux/vm/os_linux.cpp [hunk 1]
```diff
@@ -581,6 +580,13 @@ void os::Linux::hotspot_sigmask(Thread* thread) {
 // detecting pthread library
 
 void os::Linux::libpthread_init() {
+#if 1
+  // Hard code Alpine Linux supported musl compatible settings
```
Comment (jerboaa) [inline hotspot/src/os/linux/vm/os_linux.cpp]: Is this intentional? How is this Alpine Linux specific with `#if 1`? If not Alpine specific, why hard-code to `glibc 2.9`?

**Comment by gdams** [comment]
For those of you in the Temurin Compliance Group, I'll be running some TCK tests against these patches to verify that they are in good shape: https://github.com/temurin-compliance/temurin-compliance/issues/136

**Review by jerboaa — COMMENTED** [review]
I haven't reviewed thoroughly. It seems fine from a quick glance.

File: hotspot/src/os/linux/vm/os_linux.cpp (hunk 1 above)
Comment (calderast) [reply hotspot/src/os/linux/vm/os_linux.cpp]: Fixed, thank you for pointing that out!

File: hotspot/src/share/vm/utilities/bitMap.cpp [hunk 2]
```diff
@@ -42,6 +42,7 @@
 # include "os_bsd.inline.hpp"
 #endif
 
+template class ArrayAllocator<BitMap::bm_word_t, mtInternal>;
```
Comment (karianna) [inline hotspot/src/share/vm/utilities/bitMap.cpp:45]: Is this really required?

**Comment by karianna** [comment]
Overall LGTM - I could push for a few more comments but I think keeping the majority of the patch matching Alpine and IcedTea upstream will mean less clashes later when this gets proposed in jdk8u by Bellsoft.

**Comment by karianna** [comment]
Holding off merging until TCK passes

**Comment by llamahunter** [comment]
ah... this is the long awaited java 8 alpine docker image musl support? Looks like it's on its way for release sometime soon, correct?

**Comment by karianna** [comment]
That is correct.  It'll be offered by Adoptium to start with but we're submitting it as a proposal upstream as well.  We are *hoping* for the January PSU release timeframe, but our release test pipelines will determine if we release or not.

**Pull request merged**